collections/files.go

```diff
@@ -0,0 +1,89 @@
+package collections
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/ipfs/interface-go-ipfs-core/path"
+	"github.com/textileio/go-threads/api/client"
+	s "github.com/textileio/go-threads/store"
+)
+
+type File struct {
```

Comment:
I'd like to add a `Parent` in a follow up PR, which can be used to build a directory structure.